### PR TITLE
rados: split omap functions and object Iter functions into seperate go files

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Create omap.go and object_iter.go which contain functions formerly in ioctx.go.
Add missing / fix incorrect doc comments for those files.
No functionality / significant code changes were made.

## Checklist
- (n/a) Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code

### CODE REVIEW

File: rados/object_iter.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <errno.h>
+// #include <stdlib.h>
```
Comment (nixpanic) [inline rados/object_iter.go]: errno.h and stdlib.h do not seem to be used?

File: rados/omap.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <errno.h>
```
Comment (nixpanic) [inline rados/omap.go]: errno.h does not seem to be used?

File: rados/object_iter.go (hunk 1 above)
Comment (phlogistonjohn) [reply rados/object_iter.go]: stdlib is needed for free afaict. errno can be removed.

**Comment by phlogistonjohn** [comment]
Patches updated, comments resolved.

**Pull request merged**